x-pack/test/cloud_integration/fixtures/saml/saml_provider/server/saml_tools.ts

```diff
@@ -0,0 +1,161 @@
+const deflateRawAsync = promisify(zlib.deflateRaw);
+const parseStringAsync = promisify(parseString);
+
+const signingKey = fs.readFileSync(KBN_KEY_PATH);
+const signatureAlgorithm = 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha256';
+
+export async function getSAMLRequestId(urlWithSAMLRequestId: string) {
+  const inflatedSAMLRequest = (await inflateRawAsync(
+    Buffer.from(
+      url.parse(urlWithSAMLRequestId, true /* parseQueryString */).query.SAMLRequest as string,
+      'base64'
+    )
+  )) as Buffer;
+
+  const parsedSAMLRequest = (await parseStringAsync(inflatedSAMLRequest.toString())) as any;
+  return parsedSAMLRequest['saml2p:AuthnRequest'].$.ID;
+}
+
+export async function getSAMLResponse({
+  destination,
+  inResponseTo,
+  sessionIndex,
+  username = 'a@b.c',
+  issuer = 'http://www.elastic.co/saml1',
+}: {
+  destination?: string;
+  inResponseTo?: string;
+  sessionIndex?: string;
+  username?: string;
+  issuer?: string;
+} = {}) {
+  const issueInstant = new Date().toISOString();
+  const notOnOrAfter = new Date(Date.now() + 3600 * 1000).toISOString();
+
+  const samlAssertionTemplateXML = `
+    <saml:Assertion xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion" Version="2.0"
+                    ID="_RPs1WfOkul8lZ72DtJtes0BKyPgaCamg" IssueInstant="${issueInstant}">
+      <saml:Issuer>${issuer}</saml:Issuer>
+      <saml:Subject>
+        <saml:NameID Format="urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified">a@b.c</saml:NameID>
+        <saml:SubjectConfirmation Method="urn:oasis:names:tc:SAML:2.0:cm:bearer">
+          <saml:SubjectConfirmationData NotOnOrAfter="${notOnOrAfter}"
+                                        Recipient="${destination}"
+                                        ${inResponseTo ? `InResponseTo="${inResponseTo}"` : ''} />
+        </saml:SubjectConfirmation>
+      </saml:Subject>
+      <saml:AuthnStatement AuthnInstant="${issueInstant}" SessionIndex="${sessionIndex}">
+        <saml:AuthnContext>
+          <saml:AuthnContextClassRef>urn:oasis:names:tc:SAML:2.0:ac:classes:unspecified</saml:AuthnContextClassRef>
+        </saml:AuthnContext>
+      </saml:AuthnStatement>
+      <saml:AttributeStatement xmlns:xs="http://www.w3.org/2001/XMLSchema"
+                             xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+        <saml:Attribute FriendlyName="principal" Name="http://saml.elastic-cloud.com/attributes/principal" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:uri">
+          <saml:AttributeValue xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="xs:string">${CLOUD_USER_ID}</saml:AttributeValue>
+        </saml:Attribute>
+        <saml:Attribute FriendlyName="roles" Name="http://saml.elastic-cloud.com/attributes/roles" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:uri">
+          <saml:AttributeValue xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="xs:string">superuser</saml:AttributeValue>
+        </saml:Attribute>
```

Comment:
These attributes match what Cloud SSO sends using the fully qualified attribute names